contracts/utils/structs/MerkleTree.sol

```diff
@@ -19,6 +19,11 @@ import {Panic} from "../Panic.sol";
  * * Zero value: The value that represents an empty leaf. Used to avoid regular zero values to be part of the tree.
  * * Hashing function: A cryptographic hash function used to produce internal nodes. Defaults to {Hashes-commutativeKeccak256}.
  *
+ * proving the inclusion of a leaf is not possible with the {MerkleProof} library since it only supports
+ * NOTE: Building trees using non-commutative hashing functions (i.e. `H(a, b) != H(b, a)`) is supported. However,
```

Comment:
```suggestion
 * NOTE: Building trees using non-commutative hashing functions (i.e. `H(a, b) != H(b, a)`) is supported. However,
 * proving the inclusion of a leaf is not possible with the {MerkleProof} library since it only supports
```